apps/nextjs/src/stores/moderationStore/index.ts

```diff
@@ -0,0 +1,88 @@
+import type { PersistedModerationBase } from "@oakai/core/src/utils/ailaModeration/moderationSchema";
+import type { Moderation } from "@prisma/client";
+import { create } from "zustand";
+
+import type { TrpcUtils } from "@/utils/trpc";
+
+import type { ChatStore } from "../chatStore";
+import { logStoreUpdates } from "../zustandHelpers";
+import { handleFetchModerations } from "./actionFunctions/handleFetchModeration";
+import { handleToxicModeration } from "./actionFunctions/handleToxicModeration";
+import { handleUpdateModerationState } from "./actionFunctions/handleUpdateModerationState";
+
+export type ModerationStore = {
+  chatActions?: Pick<ChatStore, "setMessages">;
+
+  id: string | null;
+  moderations: Moderation[] | [];
+  isModerationsLoading: boolean | null;
+  toxicInitialModeration: PersistedModerationBase | null;
+  toxicModeration: PersistedModerationBase | null;
+  lastModeration: PersistedModerationBase | null;
+
+  updateToxicModeration: (mod: PersistedModerationBase | null) => void;
+  setLastModeration: (mod: PersistedModerationBase | null) => void;
+  setIsModerationsLoading: (isModerationsLoading: boolean) => void;
+  updateModerationState: (mods?: Moderation[]) => void;
+  fetchModerations: () => Promise<void>;
+  reset: (params: Partial<ModerationStore>) => void;
+  clearModerations: () => void;
+};
+
+export const createModerationStore = ({
+  id = null,
+  initialValues = {},
+  trpcUtils,
+}: {
+  id?: string | null;
```

Comment:
Should we just always expect this? Or do you think that it would help with storybook and testing to make it optional?